Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1

```diff
@@ -1625,6 +1625,303 @@ function Test-DiscoveryService{
+    $existingAzureChildNodes = Get-XMLChildNodes -installConfigPath $azureConfigPath -configSection $configSection -nodesToSearch $nodesToSearch -childNodeGetAttribute $childNodeGetAttribute
+    
+    # Check if install.config and azuresettings.config AAD settings match
+    $areSame = Compare-Object -ReferenceObject $existingChildNodes -DifferenceObject $existingAzureChildNodes -Property name, alias, appName, tenantId, tenantAlias, clientid, secret
+    if($null -ne $areSame)
+    {
+      Write-DosMessage -Level "Information" -Message "Since install.config and azuresettings.config AAD Settings are different, install.config settings were not deleted in the upgrade to azuresettings.config"
+    }
+    else
+    {
+      # Remove AAD Settings from install.config
+      Remove-XMLChildNodes -azureConfigPath $installConfigPath -configSection $configSection -nodesToSearch $nodesToSearch -childNodeGetAttribute $childNodeGetAttribute -childNodeAttributeSetting "" -reverseOperator "not equal"
+      Remove-XMLChildNode -azureConfigPath $installConfigPath -configSection $configSection -childNodeGetAttribute $childNodeGetAttribute -childNodeAttributeSetting "azureSecretName"
+    }
+    return $true
+}
+
+function Remove-XMLChildNode {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $azureConfigPath,
+        [Parameter(Mandatory=$true)]
+        [string] $configSection,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeGetAttribute,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeAttributeSetting
+    )
+    # Clean out empty variables in default azureSettings.config
+    $azureInstallationConfig = [xml](Get-Content $azureConfigPath)
+    $azureIdentityScope = $azureInstallationConfig.installation.settings.scope | Where-Object {$_.name -eq $configSection}
+
+    $removeEmptyVariable = $azureIdentityScope.ChildNodes | Where-Object {$_.$childNodeGetAttribute -eq $childNodeAttributeSetting}
+    $removeEmptyVariable.ParentNode.RemoveChild($removeEmptyVariable)
+
+    $azureInstallationConfig.Save("$azureConfigPath") | Out-Null
+}
+
+function Remove-XMLChildNodes {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $azureConfigPath,
+        [Parameter(Mandatory=$true)]
+        [string] $configSection,
+        [Parameter(Mandatory=$true)]
+        [string[]] $nodesToSearch,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeGetAttribute,
+        [string] $childNodeAttributeSetting,
+        [string] $reverseOperator
+    )
+    # Clean out empty variables in default azureSettings.config
+    $azureInstallationConfig = [xml](Get-Content $azureConfigPath)
+    $azureIdentityScope = $azureInstallationConfig.installation.settings.scope | Where-Object {$_.name -eq $configSection}
+
+    foreach($nodeToSearch in $nodesToSearch)
+    {
+     $setAzureSettings = $azureIdentityScope.SelectSingleNode($nodeToSearch)
+
+     if($reverseOperator -eq "not equal")
```

Comment:
if I'm understanding this, if this is set to "not equal" then it removes all other nodes otherwise it removes only the nodesToSearch's children?

Reply:
If it is set to "" then it removes the default azuresettings.config <variable name = "" alias = ""/> for each nodeToSearch. If it is set to "not equal" then it removes the <variable name="something" alias = "something" /> currently in install.config, after migrating settings to azuresettings.config for each nodeToSearch.

Reply:
If the reverseoperator is not set, then it removes the default azuresettings.config blank <variable name = "" alias = ""/> for each nodeToSearch, where $childNodeAttributeSetting -eq "". If the reverseoperator is set to "not equal" then it removes the non-blank <variable name="something" alias = "something" /> currently in install.config, for each nodeToSearch, where $childNodeAttributeSetting -ne "".